lib/zendesk_apps_support/assets/src.js.erb

```diff
@@ -1,32 +1,35 @@
+<% unless iframe_only %>
 with( ZendeskApps.AppScope.create() ) {
   <% unless modules.empty? %>
-    require.modules = {
+  require.modules = {
       <% modules.each do |path, content| %>
-        <%= path.to_json %>: function(exports, require, module) {
-          <%= content %>
-        },
+      <%= path.to_json %>: function(exports, require, module) {
+        <%= content %>
+      },
       <% end %>
-      eom: undefined
-    };
+    eom: undefined
+  };
   <% end %>
 
   var source = <%= source %>;
-
-  var app = ZendeskApps.defineApp(source)
-    .reopenClass(<%= app_settings.to_json %>)
-    .reopen({
-      appName: <%= name.to_json %>,
-      appVersion: <%= version.to_json %>,
-      assetUrlPrefix: <%= asset_url_prefix.to_json %>,
-      appClassName: <%= app_class_name.to_json %>,
-      author: {
-        name: <%= author['name'].to_json %>,
-        email: <%= author['email'].to_json %>
-      },
-      translations: <%= translations.to_json %>,
-      templates: <%= templates.to_json %>,
-      frameworkVersion: <%= framework_version.to_json %>
-    });
-
-  ZendeskApps[<%= name.to_json %>] = app;
 }
+<% end %>
+var app = ZendeskApps.defineApp(<%= iframe_only ? 'null' : 'source' %>)
```

Comment:
Perhaps we could define `source` at the top and leave it `undefined` for `iframe_only`?